File: src/main/java/be/superjoran/mint/domain/Category.java
package be.superjoran.mint.domain;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

/**
 * Created by Jorandeboever
 * Date: 01-Oct-16.
 */
@Entity
@Table(name = "T_CATEGORY")
public class Category extends DomainObject {
    private static final long serialVersionUID = -163498981511624588L;

    @ManyToOne(optional = false)
    @JoinColumn(name = "CATEGORY_GROUP_UUID")
    private CategoryGroup categoryGroup;

    private String name;

    public Category() {
    }

    public CategoryGroup getCategoryGroup() {
        return this.categoryGroup;
    }

    public void setCategoryGroup(CategoryGroup categoryGroup) {
        this.categoryGroup = categoryGroup;
    }

    public Category(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String getDisplayValue() {
        return this.toString();
    }

    @Override
    public String toString() {
        return String.format("%s - %s", this.categoryGroup.getName(), this.name);
    }
}

File: src/main/java/be/superjoran/mint/domain/Category.kt
package be.superjoran.mint.domain

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

/**
 * Created by Jorandeboever
 * Date: 01-Oct-16.
 */
@Entity
@Table(name = "T_CATEGORY")
class Category(
        uuid: String,
        var name: String,
        @ManyToOne(optional = false)
        @JoinColumn(name = "CATEGORY_GROUP_UUID")
        var categoryGroup: CategoryGroup
) : DomainObject(uuid) {

    override fun toString(): String {
        return String.format("%s - %s", this.categoryGroup.name, this.name)
    }

}
